components/director/pkg/resource/resource.go

```diff
@@ -1,77 +1,179 @@
+	// APISpecification type represents API specification resource.
+	APISpecification Type = "apiSpecification"
+	// EventSpecification type represents Event specification resource.
+	EventSpecification Type = "eventSpecification"
+	// Document type represents document resource.
 	Document Type = "document"
-	// BundleInstanceAuth missing godoc
+	// BundleInstanceAuth type represents bundle instance auth resource.
 	BundleInstanceAuth Type = "bundleInstanceAuth"
-	// API missing godoc
+	// API type represents api resource.
 	API Type = "api"
-	// EventDefinition missing godoc
+	// EventDefinition type represents event resource.
 	EventDefinition Type = "eventDefinition"
-	// AutomaticScenarioAssigment missing godoc
+	// AutomaticScenarioAssigment type represents ASA resource.
 	AutomaticScenarioAssigment Type = "automaticScenarioAssigment"
-	// Webhook missing godoc
+	// Webhook type represents generic webhook resource. This resource does not assume the referenced resource type of the Webhook.
 	Webhook Type = "webhook"
-	// Tenant missing godoc
+	// AppWebhook type represents application webhook resource.
+	AppWebhook Type = "appWebhook"
+	// RuntimeWebhook type represents runtime webhook resource.
+	RuntimeWebhook Type = "runtimeWebhook"
+	// Tenant type represents tenant resource.
 	Tenant Type = "tenant"
-	// TenantIndex missing godoc
-	TenantIndex Type = "tenantIndex"
-	// Schema missing godoc
+	// TenantAccess type represents tenant access resource.
+	TenantAccess Type = "tenantAccess"
+	// Schema type represents schema resource.
 	Schema Type = "schemaMigration"
 )
 
-// SQLOperation missing godoc
+var tenantAccessTable = map[Type]string{
+	// Tables
+
+	Application: "tenant_applications",
+	Runtime:     "tenant_runtimes",
+
+	// Views
+
+	RuntimeContext:        "runtime_contexts_tenants",
+	Label:                 "labels_tenants",
+	ApplicationLabel:      "application_labels_tenants",
+	RuntimeLabel:          "runtime_labels_tenants",
+	RuntimeContextLabel:   "runtime_contexts_labels_tenants",
+	Bundle:                "bundles_tenants",
+	Package:               "packages_tenants",
+	Product:               "products_tenants",
+	Vendor:                "vendors_tenants",
+	Tombstone:             "tombstones_tenants",
+	DocFetchRequest:       "document_fetch_requests_tenants",
+	APISpecFetchRequest:   "api_specifications_fetch_requests_tenants",
+	EventSpecFetchRequest: "event_specifications_fetch_requests_tenants",
+	APISpecification:      "api_specifications_tenants",
+	EventSpecification:    "event_specifications_tenants",
+	Document:              "documents_tenants",
+	BundleInstanceAuth:    "bundle_instance_auths_tenants",
+	API:                   "api_definitions_tenants",
+	EventDefinition:       "event_api_definitions_tenants",
+	Webhook:               "webhooks_tenants",
+	AppWebhook:            "application_webhooks_tenants",
+	RuntimeWebhook:        "runtime_webhooks_tenants",
+}
+
+var parentRelation = map[Type]Type{
```

Comment:
this mapping you can probably autodiscover via the information schema, https://www.postgresql.org/message-id/40C500D8.9060009@actarg.com where you can explore contraints, fks and metadata about them

Reply:
Yes, of course you can, but does it give us anything to be worth it the complexity?